pkg/operator/configobservation/node/observe_latency_profile_test.go

```diff
@@ -0,0 +1,102 @@
+package node
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/configobservation"
+)
+
+type workerLatencyProfileTestScenario struct {
+	name                   string
+	existingConfig         map[string]interface{}
+	expectedObservedConfig map[string]interface{}
+	workerLatencyProfile   configv1.WorkerLatencyProfileType
+}
+
+func multiScenarioLatencyProfilesTest(
+	configObserveFn func(configobserver.Listers, events.Recorder, map[string]interface{}) (ret map[string]interface{}, errs []error),
+	scenarios []workerLatencyProfileTestScenario,
+) func(*testing.T) {
+	return func(t *testing.T) {
+		for _, scenario := range scenarios {
+			t.Run(scenario.name, func(t *testing.T) {
+				// test data
+				eventRecorder := events.NewInMemoryRecorder("")
+				configNodeIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+				configNodeIndexer.Add(&configv1.Node{
+					ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+					Spec:       configv1.NodeSpec{WorkerLatencyProfile: scenario.workerLatencyProfile},
+				})
+				listers := configobservation.Listers{
+					NodeLister: configlistersv1.NewNodeLister(configNodeIndexer),
+				}
+
+				// act
+				observedKubeAPIConfig, err := configObserveFn(listers, eventRecorder, scenario.existingConfig)
```

Comment:
what is the benefit of passing the function?

Reply:
Updated to use `ObserveNodeMonitorGracePeriod` call directly here.